netty-reactive-streams/src/main/java/com/typesafe/netty/HandlerPublisher.java

```diff
@@ -203,32 +203,40 @@ private void provideSubscriber(Subscriber<? super T> subscriber) {
 
     @Override
     public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
-        verifyRegisteredWithRightExecutor(ctx);
+        // If the channel is not yet registered, then it's not safe to invoke any methods on it, eg read() or close()
+        // So don't provide the context until it is registered.
+        if (ctx.channel().isRegistered()) {
+            provideChannelContext(ctx);
+        }
```

Comment:
No, because if you add the handler to the pipeline after the channel has registered (which is what we do in Play when handling request/response bodies), you'll never receive the `channelRegistered` event.